lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -1,38 +1,78 @@
 import { createElement } from 'preact';
 import classNames from 'classnames';
-import propTypes from 'prop-types';
 import { useContext, useEffect, useState, useRef } from 'preact/hooks';
+import propTypes from 'prop-types';
 
 import { Config } from '../config';
-import { submitGrade } from '../utils/grader-service';
-import { formatToNumber, validateGrade } from '../utils/validation';
 import ErrorDialog from './ErrorDialog';
 import Spinner from './Spinner';
 import SvgIcon from './SvgIcon';
+import { fetchGrade, submitGrade } from '../utils/grader-service';
 import { trustMarkup } from '../utils/trusted';
+import { formatToNumber, validateGrade } from '../utils/validation';
 import ValidationMessage from './ValidationMessage';
 
+// Grades are always stored as a value between [0-1] on the service layer.
+// Scale the grade to [0-10] when loading it into the UI and scale it
+// back to [0-1] when saving it to the api service.
+const GRADE_MULTIPLIER = 10;
+
+/**
+ * Custom useEffect function that handles fetching a student's
+ * grade and returning the result of that grade and the loading
+ * and error status of that fetch request.
+ */
+const useFetchGrade = student => {
+  const { authToken } = useContext(Config);
+  const [grade, setGrade] = useState('');
+  const [gradeLoading, setGradeLoading] = useState(false);
+
+  useEffect(() => {
+    if (Object.entries(student).length) {
+      // Don't bother fetching if the student object is empty
+      // Fetch the grade from the service api
+      // See https://www.robinwieruch.de/react-hooks-fetch-data for async in useEffect
+      const fetchData = async () => {
+        setGradeLoading(true);
+        setGrade(''); // Clear previous grade so we don't show the wrong grade with the new student
+        const response = await fetchGrade({ student, authToken });
+        setGrade(response.currentScore * GRADE_MULTIPLIER);
+        setGradeLoading(false);
```

Comment:
I believe there is a race condition here if:

1. The component loads and the grade is fetched for Student 1
2. The user switches to Student 2
3. The grade is fetched for Student 2
4. The response comes back for Student 2
5. The response from (1) comes back after (4)

A possible solution would be to return a cleanup function that cancels the current run by setting a flag which is checked before calling `setGrade`/`setGradeLoading` with the response.